### PR TITLE
Fixing Solution to Telephone Validation Problem

### PR DESCRIPTION
**Pull request by matthew-t-smith**
4 instances of an escaped backslash + s ("\\s") that should simply be a whitespace character match ("\s") fixed. Additional backslashes removed from test at end of solution that escaped quotations around the "555-555-5555" number.

I re-verified that the previous solution failed 4 of the tests (2, 3, 11 and 12), but the new solution solves all the tests on the live site.

#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [ ] Closes currently open issue (replace XXXX with an issue no): Closes #XXXX

### CODE REVIEW

File: seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-algorithm-questions.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
         "telephoneCheck(\"555-555-5555\");"
       ],
       "solutions": [
-        "var re = /^([+]?1[\\s]?)?((?:[(](?:[2-9]1[02-9]|[2-9][02-8][0-9])[)][\\s]?)|(?:(?:[2-9]1[02-9]|[2-9][02-8][0-9])[\\s.-]?)){1}([2-9]1[02-9]|[2-9][02-9]1|[2-9][02-9]{2}[\\s.-]?){1}([0-9]{4}){1}$/;\n\nfunction telephoneCheck(str) {\n  return re.test(str);\n}\n\ntelephoneCheck(\"555-555-5555\");"
+        "var re = /^([+]?1[\s]?)?((?:[(](?:[2-9]1[02-9]|[2-9][02-8][0-9])[)][\s]?)|(?:(?:[2-9]1[02-9]|[2-9][02-8][0-9])[\s.-]?)){1}([2-9]1[02-9]|[2-9][02-9]1|[2-9][02-9]{2}[\s.-]?){1}([0-9]{4}){1}$/;\n\nfunction telephoneCheck(str) {\n  return re.test(str);\n}\n\ntelephoneCheck("555-555-5555");"
```
Comment (dhcodes) [inline seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-algorithm-questions.json]: The extra backslash on an escaped regex is due to how the test structure handles the JSON file where the challenges and tests are stored. The escaped quotes at the end are necessary for similar reasons, I believe. Was this causing an issue on the challenge in any way?

**Review by ahshame8657 — APPROVED** [review]

File: seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-algorithm-questions.json (hunk 1 above)
Comment (matthew-t-smith) [reply seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-algorithm-questions.json]: I was unaware of the extra escapes needed for JSON. Does this become necessary then for the `\n`s throughout?

Comment (dhcodes) [reply seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-algorithm-questions.json]: @matthew-t-smith yes it should apply to all escaped characters in regex or js.

**Comment by systimotic** [comment]
To add to what @dhcodes already said, the double escaping is expected and required.

When this is parsed from JSON into Javascript, the backslashes are removed for the first time. It will read `\s` as an escaped `s` and parse it into an `s`. It will read `\\s` as an escaped `\` and parse it into `\s`, which is what we want.

To test the solution in the browser, it should first be parsed into Javascript using `JSON.parse`. Our test suite does this automatically. As you can see, running `npm test` was successful before, but is failing now. This is why the red `x` is showing up on your PR.

I'm closing this PR. We'd welcome your other contributions! If we have confirmed it needs to be resolved, an issue will be labeled ["help wanted"](https://github.com/freeCodeCamp/freeCodeCamp/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22)


Happy coding!

**Pull request closed without merging**